controllers/media.js

```diff
@@ -0,0 +1,121 @@
+  const path = mediaUrl.substr(mediaUrl.indexOf('/media') + offset, mediaUrl.length)
+
+  return path
+}
+
+const uploadMedia = async (req, res) => {
+  if (!req.formdata || !req.formdata.files) {
+    return res.status(HTTPStatus.BAD_REQUEST).send({ errors: ['no files uploaded'] })
+  }
+
+  const files = []
+  const badTypes = []
+  for (const key in req.formdata.files) {
+    if (key === '') return res.status(HTTPStatus.BAD_REQUEST).send({ errors: ['Add key for each file of the request'] })
+    const file = req.formdata.files[key]
+    if (file.type.split('/')[0] !== 'image') {
+      badTypes.push(file.name)
+    }
+    files.push(file)
+  }
+
+  if (badTypes.length) {
+    return res.status(HTTPStatus.BAD_REQUEST).send({
+      errors: badTypes.map((f) => ({
+        file: f,
+        error: 'invalid type (image expected)',
+      })),
+    })
+  }
+
+  const storageClient = Storage.getStorageClient()
+  const savePromises = files.map((f) => {
+    const extension = f.name.split('.').pop()
+    return storageClient.saveFile(f.path, `media-${req.params.orgId}-${uuidv4()}.${extension}`)
+  })
+  const saveResults = await Promise.all(savePromises)
+
+  try {
+    await Promise.all(files.map((f) => fs.unlink(f.path)))
+  } catch (err) {
+    log.error(err, 'uploadMedia: failed to remove temporary files')
+    return res.status(HTTPStatus.INTERNAL_SERVER_ERROR).send('uploadMedia: failed to remove temporary files')
+  }
+
+  const response = {
+    savedImages: [],
+    errors: [],
+  }
+
+  const transaction = await sequelize.transaction()
+  for (let j = 0; j < saveResults.length; j++) {
+    const sr = saveResults[j]
+    if (sr.objectURL && sr.objectURL.length) {
+      const file = pathParser(sr.objectURL, 7)
+      response.savedImages.push({
+        file: files[j].name,
+        url: sr.objectURL,
+      })
+      await models.Media.create({
+        file: file,
+        url: sr.objectURL,
+        orgId: req.params.orgId,
+      }, transaction)
+      continue
+    }
+
+    await transaction.commit()
+
+    response.errors.push({
+      file: files[j].name,
+      error: 'failed to save image',
+    })
+  }
+
+  if (!response.errors.length) delete response.errors
+  return res.status(HTTPStatus.OK).send(response)
+}
+
+const deleteMedia = async (req, res) => {
+  const transaction = await sequelize.transaction()
+
+  const file = pathParser(req.query.mediaUrl, 7)
+  const mediaSearch = await models.Media.findByPk(file, { transaction })
+
+  if (!mediaSearch) {
+    if (transaction) await transaction.rollback()
+    return res.status(HTTPStatus.NOT_FOUND).send({ errors: ['Image not found'] })
+  }
+
+  const path = pathParser(req.query.mediaUrl, 0)
+
+  await models.Media.destroy({
+    where: {
+      file: file,
+      orgId: req.params.orgId,
+    },
+    transaction,
+  })
+  await transaction.commit()
+
+  const storageClient = Storage.getStorageClient()
+  await storageClient.deleteFile(path)
```

Comment:
Shouldn't this happen before the model destroy (and commit), therefore preventing the model to be deleted if the image delete fails?